mesonbuild/backend/vs2010backend.py

```diff
@@ -861,7 +861,14 @@ def gen_vcxproj(self, target, ofname, guid):
             if not pch:
                 continue
             pch_node.text = 'Use'
-            pch_sources[lang] = [pch[0], pch[1], lang]
+            if compiler.id == 'msvc':
+                if len(pch) != 2:
+                    raise RuntimeError('MSVC requires one header and one source to produce precompiled headers.')
```

Comment:
This should be a `MesonException` so that the error handling mechanism can do the right thing.

Reply:
I blindly copy/pasted the same block in ninjbackend.py:
```python
    def generate_msvc_pch_command(self, target, compiler, pch):
        if len(pch) != 2:
            raise RuntimeError('MSVC requires one header and one source to produce precompiled headers.')
```

I will change them following your recommandations.